Add GriefPrevention

RoboMWM opened the pull request:
- Provides self-service claims (I changed the claim tool from a gold shovel to a wood shovel so it's more easily accessible.)
- Handles spammers properly via a "softmute" effect, where only they see their own spam. NCP lets spam flood chat, only taking action via kicking when it has reached a certain threshold, while GP silently "softmutes" spam until the player has stopped spamming (or gets themselves kicked for macrospam). Filters gibberish, repeated messages, macrospam, and IP addresses (not domains)
- PvP rules for non-PvP worlds: pet & horse protection, disallow placing fire/lava near players, protects drops on death (can toggle via player and config)
- Smarter ban system which auto-bans players who have the same IP as the recently-banned player. (lasts only a day, as IPs can easily change)

https://www.spigotmc.org/resources/griefprevention.1884

RoboMWM committed:
Add GriefPrevention

RoboMWM commented:
Configured to have minimal impact to anything else. Only exception is if a plotworld plugin creates a world, might need to disable claim function for that world in GP's config. Will PR a change to use GP's spam prevention instead of NCP if this is merged.

JamieSinn commented:
As said before, is this simply used for claiming land? If so, this really shouldn't be used. A better claim land plugin should. If that is the case, :-1:

RoboMWM committed:
Update config.yml

Uses wood-spade for claiming so players have easier access to this feature (default is gold spade)

ghost commented:
:-1: No claims needed

RoboMWM commented:
No - used for PvP grief, spam (although I disabled here unless you want me to disable NCP spam) as well. Will disable claims if not needed

RoboMWM committed:
Disabled claims

ghost commented:
PVP Is already disabled, as mentioned on the forums.

JamieSinn commented:
WG/NCP handles everything that is proposed that this handles.

RoboMWM commented:
Also handles PvP rules for non-PvP, such as pet & horse protection, not placing fire/lava near players, etc.

Handles spammers better via a "softmute" effect, where they only see their own spam.

RoboMWM committed:
Use GriefPrevention spam protection instead of NCP

GP is much more effective with spam protection, using a "softmute"
effect, where only the spammer sees his own messages (thinking they went
through instead of trying to find a way around it).

ctian1 commented:
This plugin allows players to claim their own land, as opposed to WG

it also offers many other useful anti-grief/admin related features that can be found [here](https://www.spigotmc.org/resources/griefprevention.1884/)

RoboMWM commented:
I'm not sure if self-serve claims are wanted. As it stands, I have that disabled based on preliminary feedback here.

However, it provides robust spam protection and enforces non-PvP rules such as pet/horse protection and lava/fire placement near players.

ctian1 commented:
Why not use self-serve claims?

RoboMWM commented:
Not sure.

>A better claim land plugin should [be used].

Curious to know how this is an inferior claim plugin, considering it's been around for 2 years and highly active..

ctian1 commented on plugins/GriefPreventionData/config.yml:
```diff
@@ -0,0 +1,117 @@
+    WarningMessage: Please reduce your noise level.  Spammers will be punted.
+    BanOffenders: false
+    BanMessage: Punted for spam.
+    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
```
...

Janmm14 replied:
What about adding for example 8.8.8.8 or 12.7.0.0.1?

RoboMWM replied:
This is just the default whitelist for IP addresses to be allowed in chat (does not block domains).

ctian1 commented on plugins/GriefPreventionData/config.yml:
```diff
@@ -0,0 +1,117 @@
+    PunishLogout: true
+    CombatTimeoutSeconds: 10
+    AllowCombatItemDrop: false
+    BlockedSlashCommands: /home;/vanish;/spawn;/tpa
```
vanish?

RoboMWM replied:
These are part of default config. Can be altered if needed.

ctian1 replied:
I know, I'm saying it should probably be edited

RoboMWM replied:
Doesn't matter since no PvP anyways.

ctian1 replied:
Should the PvP section just be removed?

RoboMWM replied:
'RulesEnabledInWorld' also provides pet/horse protection and flint and steel/lava dumping protection. Plugin will just regen config keys anyways.

ctian1 commented on plugins/GriefPreventionData/config.yml:
```diff
@@ -0,0 +1,117 @@
+      PlayerOwnedClaims: true
+      AdministrativeClaims: true
+      AdministrativeSubdivisions: true
+    AllowLavaDumpingNearOtherPlayers: true
```
I thought these would be disabled

RoboMWM replied:
Only affects PvP-enabled worlds.

ctian1 replied:
My bad.

jeff142 commented:
:-1:

ghost commented:
:-1: We don't need claims.

RoboMWM commented:
Claims are disabled, this isn't the only thing it does...

Re-emphasized antispam in description. NCP does not do much about spam except for extreme cases, and only via a kick. It does not even attempt to mute stuff that is clearly spam and knows is spam because chat antispam is not the primary focus of NCP.

md-5 commented:
Some sort of claiming needed :+1:

RoboMWM committed:
Revert "Disabled claims"

This reverts commit ab1df90617e924270500f9c480a798689a4c638d.

Maximvdw commented:
I like some way of claiming or protecting,  but not griefprevention for protection :-1:

RoboMWM commented:
What part about it do you not like?

TheGrimsey commented:
:+1: I like it.

Cldfire commented:
:-1:
I'd prefer districts to this https://www.spigotmc.org/resources/districts.989/

Very user-friendly, GUI based, great dev.

RoboMWM commented:
Only thing I see going for that is if people want to micromanage their claims with a chest GUI, that could do (I'd suppose a simple /trust would suffice), and I'll just disable claims from this plugin (as districts does not provide any of the other features mentioned such as anti-spam).

Otherwise, seems to add extra unnecessary things (entrance and exit messages) and lots of economy stuff. Also unknown if more blocks can be acquired other than purchasing (GP adds via non-AFK playtime, and I bumped this up a bit from the default).

HappyHarm commented:
Maybe someone can create a list with claiming plugins where we can choose about? Then we can better compare them and see which one is the most useful?

RoboMWM commented:
That'd be helpful because the only feedback I'm getting is "I don't like it" so I don't even know what to address.

ctian1 commented:
IMHO GriefPrevention is a more powerful but easier to use claim plugin than Districts.. you can read about GP [here](https://www.spigotmc.org/resources/griefprevention.1884/)

RoboMWM committed:
Disable claiming in creative world

Assuming PlotSquared is managing a world named "creative"

Techcable commented:
:+1: some form of claiming is neaded

dmulloy2 commented:
:+1: need claims, GriefPrevention has worked wonderfully for me.
